processor/src/main/java/com/linecorp/decaton/processor/runtime/internal/ProcessPipeline.java

```diff
@@ -38,7 +38,7 @@
 @Slf4j
 public class ProcessPipeline<T> implements AutoCloseable {
     private final ThreadScope scope;
-    private final List<DecatonProcessor<T>> processors;
+    private final List<DecatonProcessor<QuotaAwareTask<T>>> processors;
```

Comment:
As we discussed offline - there's some room to consider about this `QuotaAwareTask` approach.

Reply:
Discussed offline again.

In the first place, we should shape bursting tasks before we deserialize (extract) and enqueue them to processor queue to prevent abusive tasks dominate the CPU resource.

Hence, implementing bursting feature as DecatonProcessor no longer makes sense.
I added below changes:

- QuotaCallback signature change:
    * We shape tasks before the extraction. It means, we can't provide task/task metadata to the quota callback
- Exclude retry topic from the shaping target (for retry topic, per-key-quota doesn't apply)
    * If we want to shape retry tasks as well, we also have to wrap original (non-retry) tasks as DecatonTaskRequest when shaping to align the message schema.
    * The difficulty is, DecatonTaskRequest expects metadata, which can't be known at the quota-apply timing.
    * We can fill "dummy" metadata and produce to shaping tasks, but it causes another problem:
        - When extracting shaping tasks, we have to ignore (outer) DecatonTaskRequest's metadata (which is filled as "dummy") then.
        - However, if we ignore outer DecatonTaskRequest's metadata, it means we also will ignore "shaping tasks which originated from retry topic".
        - => We can't ignore retry tasks metadata because it contains important retry-related metadata
    * I decided to exclude retry-topic from the target for now, to avoid these messes.
- Quota application is now done by QuotaApplier, which is called before task extraction.